From: Simon McVittie <smcv@collabora.com>
Date: Sat, 12 Nov 2022 20:56:53 +0000
Subject: type-flags test: Force G_ENABLE_DIAGNOSTIC=1 to be set

Otherwise this test will succeed at build-time, but will fail when run
as an as-installed test via ginsttest-runner.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3070
---
 gobject/tests/type-flags.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gobject/tests/type-flags.c b/gobject/tests/type-flags.c
index c4813bf..e4e5018 100644
--- a/gobject/tests/type-flags.c
+++ b/gobject/tests/type-flags.c
@@ -141,6 +141,8 @@ main (int argc, char *argv[])
 {
   g_test_init (&argc, &argv, NULL);
 
+  g_setenv ("G_ENABLE_DIAGNOSTIC", "1", TRUE);
+
   g_test_add_func ("/type/flags/final", test_type_flags_final);
   g_test_add_func ("/type/flags/deprecated", test_type_flags_deprecated);
 
